trestle/common/model_utils.py

```diff
@@ -505,3 +506,94 @@ def _load_list(abs_path: Path, abs_trestle_root: Path) -> Tuple[Type[OscalBaseMo
             aliases_not_to_be_stripped.append(model_alias.split('.')[-1])
 
         return collection_model_type, collection_model_alias, instances_to_be_merged
+
+    @staticmethod
+    def parameter_to_dict(obj: Union[OscalBaseModel, str], partial: bool) -> Union[str, Dict[str, Any]]:
+        """
+        Convert obj to dict containing only string values, storing only the fields that have values set.
+
+        Args:
+            obj: The parameter or its consituent parts in recursive calls
+            partial: Whether to convert the entire param or just the parts needed for markdown header
+
+        Returns:
+            The converted parameter as dictionary
+        """
+        main_fields = ['id', 'label', 'values', 'select', 'choice', 'how_many']
+        if isinstance(obj, common.HowMany):
+            return obj.name
+        if isinstance(obj, common.Remarks) or isinstance(obj, common.ParameterValue):
+            return obj.__root__
+        # it is either a string already or we cast it to string
+        if not hasattr(obj, '__fields_set__'):
+            return str(obj)
+        # it is an oscal object and we need to recurse within its attributes
+        res = {}
+        for field in obj.__fields_set__:
+            if partial and field not in main_fields:
+                continue
+            attr = getattr(obj, field)
+            if not attr:
+                continue
+            if isinstance(attr, list):
+                if not attr:
+                    continue
```

Comment:
I think this is checked on lines 536-537?

Reply:
fixed